openmls/src/extensions/mod.rs

```diff
@@ -166,6 +166,200 @@ pub enum Extension {
     ExternalSenders(ExternalSendersExtension),
 }
 
+/// A list of extensions with unique extension types.
+#[derive(
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserialize,
+)]
+pub struct Extensions {
+    inner: Vec<Extension>,
+}
+
+impl Extensions {
+    /// Create an extension list that is empty.
+    pub fn empty() -> Self {
+        Self { inner: Vec::new() }
+    }
+
+    /// Create an extension list that contains a single extension.
+    pub fn single(extension: Extension) -> Self {
+        Self {
+            inner: vec![extension],
+        }
+    }
+
+    /// Create an extension list that contains multiple extensions.
+    ///
+    /// This function will fail when the list of extensions contains duplicate extension types.
+    pub fn multi(extensions: Vec<Extension>) -> Result<Self, InvalidExtensionError> {
```

Comment:
The name is a little weird. But since `TryInto` is implemented we don't need it.

Reply:
I do like the conversion traits a lot but its sometimes not clear that these are implemented. That's why I wanted to provide this constructor. Given that we have `empty` and `single`, I think it is reasonable for a user to expect another one for `multi`?